src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -48,6 +48,80 @@ static class BitOperations
             19, 27, 23, 06, 26, 05, 04, 31
         };
 
+        /// <summary>
+        /// Evaluate whether a given integral value is a power of 2.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsPow2(int value)
+        {
+            if (Popcnt.IsSupported)
+            {
+                // 1 set bit means number is a power of 2 unless it is the sign bit, so get rid of that
+                // Cast to unsigned type so logic shift not signed arithmetic shift
+                return PopCount((uint)value << 1) == 1;
```

Comment:
How is the performance of PopCount vs. [Bmi1.ResetLowestSetBit](https://docs.microsoft.com/dotnet/api/system.runtime.intrinsics.x86.bmi1.resetlowestsetbit?view=net-5.0#System_Runtime_Intrinsics_X86_Bmi1_ResetLowestSetBit_System_UInt32_)?

(Intel's BSLR instruction even stores enough information to the flags register that the result of IsPow2(uint) could be computed solely from that:

- CF = 1 if value == 0
- ZF = 1 if (value & (value - 1)) == 0
- SF = sign bit of (value & (value - 1))
- OF = 0 always

so, IsPow2(uint) would need to test ZF && !CF. It seems each flag would have to be tested with a separate instruction, as the condition codes that test multiple flags do not fit this combination. IsPow2(int) would additionally have to check for negative input.)

Reply:
`popcnt` is:
* Skylake - Latency 3, Throughput 1.0, 1 Micro-op on Port 1
* Ryzen - Latency 1, Throughput 0.25, 1 Micro-op on ALU

`blsr` is:
* Skylake - Latency 1, Throughput 0.5, 1 Micro-op on Port 1 or 5
* Ryzen - Latency 1, Throughput 0.5, 2 Micro-ops on ALU

So while it will be better for Skylake, it may not be better for Ryzen. `popcnt` is also available on more hardware overall (circa 2008, rather than circa 2013) and so is likely a better choice